node/src/components/fetcher.rs

```diff
@@ -263,6 +292,8 @@ impl ItemFetcher<Deploy> for Fetcher<Deploy> {
 }
 
 impl ItemFetcher<Block> for Fetcher<Block> {
+    const SAFE_TO_RESPOND_TO_ALL: bool = false;
```

Comment:
Why not? Isn't this by hash?

Reply:
So, heres the block data structure:

```rust
pub struct Block {
    hash: BlockHash,
    header: BlockHeader,
    body: BlockBody,
}
```

We have a couple rules here:

   - `hash == header.hash()`
   - `body.hash() == header.body_hash`
   
If these are violated then the peer is faulty...

Reply:
Then shouldn't it also be `false` for deploys? E.g. their `ttl` could exceed the maximum as specified in the chainspec.

Reply:
Sorry for the confusion! It's correct, of course: It only needs to be `false` for item types where a matching hash is insufficient to respond to everyone.
If someone sends us a block with the correct hash, it _is_ the block we asked for, even if it's invalid.